### ISSUE TITLE
Bug 58167213: [React Native New Architecture]: Incorrect name property and control type is defined for all the visualized list items.

### ISSUE THREAD

**Issue by vineethkuttan** [microsoft/react-native-gallery#599]
**Build Details:**
OS Version: Dev (OS Build 27881.1000)
App: React Native Gallery Experimental
AT: Screenreader

 

**Repro-Steps:**

1. Launch the React Native Gallery- Experimental app.
2. Navigate to the 'Expand menu' button and invoke it.
3. Navigate to 'All samples' button and invoke it.
4. Navigate to 'VisualizedList' and invoke it.
5. Navigate to all the visualized list items.
6. Observe the screen reader announcement.

**Actual Result:**
Name property is not defined and control type is defined as 'group'. And on navigating, Narrator is announcing as "group".

**Expected Result:**
Name property should be defined and control type should be defined as "list items". And while navigating with arrow keys narrator should announces as "Item 12, list items, <count & position>, <selection state if selected>". All the list items should be navigable with arrow keys.

User impact:

Users with visual impairment who rely on screen reader will face difficulties and might get confused if Narrator stays only announces as group.

MAS Reference: [MAS 4.1.2 – Name, Role, Value](https://aka.ms/MAS4.1.2)

Trap ID: [1.6 - Poor Grouping](https://aka.ms/PoorGrouping)
